src/main/java/org/schemaspy/SchemaAnalyzer.java

```diff
@@ -73,6 +73,8 @@ public SchemaAnalyzer(SqlService sqlService, DatabaseService databaseService, Co
         this.databaseService = Objects.requireNonNull(databaseService);
         this.commandLineArguments = Objects.requireNonNull(commandLineArguments);
         addOutputProducer(new XmlProducerUsingDOM());
+        File outputDirectory = commandLineArguments.getOutputDirectory();
+        Dot.setInstance(commandLineArguments.isRenderDotInJvm(), outputDirectory!=null?outputDirectory.getPath():null);
```

Comment:
outputDirectory is a required parameter you shouldn't need check it for null
But since we bring the spring context up even in cases when we aren't going to use outputDir such as -help -dbhelp you'll need the null check.

I'm gonna think about that, would be nice to no require null checks on required parameters.

Reply:
Too many tests failed without this check:
[ERROR] Tests run: 37, Failures: 0, Errors: 25, Skipped: 6

Reply:
Yeah, figured it. Only way to avoid it would be to use output Directory later.

So that the getPath is only called if it's actually going to be used.

Since all constructors will be called, during spring context startup even if they won't be used.
One might argue that dot stuff shouldn't be initialized if -nohtml. And the 37 test that fail should have -nohtml.